### DISCUSSION TITLE
Discussion about future pricing

### DISCUSSION THREAD

**Discussion by nym21** [supabase/supabase#521] [Questions] [answered]
Hi !

I have been looking for many many (many...) hours for something like **Supabase**, I have yet to try it fully but on paper it looks **really promising** and it being free during development and for 1 year after its release is simply amazing and very generous ! I really appreciate that.

That being said, I truly wish we had a bit more insight about the pricing, even if it's subject to change. I have many questions which I'm sure would interest many and feel reluctant to fully switch all my projects to it before having more info:

**Do you plan to have a global plan or per database plan ?**

I keep seeing the latter (besides on Fauna DB) and I really find it counter developer friendly. Having a global plan would personally make me feel more free, by using your solution in all my projects. No matter if I need a really tiny database or an enormous beast.

**Do you plan to have a free tier ? If so would it be only for open source projects ?**

I love to create small web apps, most of the time I need a database but just a small one, nothing fancy. From what you've been doing from now, I'm sure you get it but I'll write it anyway. Having a free tier would make people want to try this database even more and use it whenever possible.

**What would be the entry cost ? Will the cost be scalable or is it going to be clear tiers ?**

I keep seeing great services which start their pricing at **15 dollars per month**. If you're a company it's cheap but again for small devs, with a tiny project or for students (which I am), it's way too much. I truly hope you have also the small guys in mind, something a bit beefier than the free tier for 5$ per month and then the next for 10$, then 15$, ..., would be great. It could also be scalable, again talking about **Fauna DB** but their pricing plan (before they changed, it's now starting at 25 dollars per month after the limited and locked free tier, which is too much IMO) is great, you have a free tier and then you pay exactly for your usage (read ops, write ops and storage).

I hope I post in the right place.

Anyway, thank you for your great work !

**Comment by kiwicopple** [comment, marked as answer]
Hey @gawlk - thanks for the very clear questions! I've just seen this because we're going back through all discussions this week. I'm sorry it took so long to answer (literally over a year 😓). Since you've written this I'm sure you've seen the pricing, but I'll answer it with the most up-to-date info.

> Do you plan to have a global plan or per database plan ?

Per database. For context, Supabase provides a full Postgres database per project (technically a "cluster"). It's not an abstraction over postgres, and it's not "postgres compatible" - it's literally "just postgres". This makes it quite expensive for us to offer a free tier! So while we would like to make the pricing plans even cheaper, we only have 2 options:

1. Build a "multi-tenant layer" over postgres, to reduce our costs
2. Stick to our current infrastructure, with the aim of lowering our prices over time (eg, hardware gets cheaper, and as we get more enterprise customers we can support more indie devs)

We're aiming for `2`, because we want don't want to abstract Postgres and lock developers into Supabase. Developers should be able to "pg_dump/pg_restore" in and out of supabase at any time they want. 

> Do you plan to have a free tier ?

Yes! https://supabase.com/pricing (the platform has a free tier and self-hosting is free too).

> I keep seeing great services which start their pricing at 15 dollars per month. If you're a company it's cheap but again for small devs, with a tiny project or for students (which I am), it's way too much. 

I understand - hopefully our free tier helps here and of course self-hosting is free. At this stage Supabase needs to build something sustainable - we don't want to fall into the same trap as other SaaS companies, where they start with a low price, get everyone hooked, and then ramp up prices after they have enough users. We hope to do the opposite - we're offering a Pro tier at a price which we feel is economically viable and over time our business economics will allow us to support more indie devs.
